### ISSUE TITLE
Errors reading S3 configuration in YAML

### ISSUE THREAD

**Issue by jpds** [quickwit-oss/quickwit#3586]
**Describe the bug**

Using v0.6.1, and the following YAML, quickwit does not appear to use the settings configured there:

```
metastore_uri: s3://qw-test/indexes
default_index_root_uri: s3://qw-test/indexes
storage:
   s3:
     access_key_id: "..."
     secret_access_key: "..."
     endpoint: https://s3.example.com
     region: "garage"
     force_path_style_access: true
     disable_multi_object_delete_requests: false
```

Note, that there are no extra `QW_` variables set at this point - running results these errors:

```
$ ./quickwit run --config quickwit.yaml
...
✘ Command failed: Failed to connect to metastore: `Internal error: `Failed to get index files.` Cause: `StorageError(kind=InternalError, source=failed to construct request: A region must be set when sending requests to S3. (ConstructionFailure(ConstructionFailure { source: ResolveEndpointError { message: "A region must be set when sending requests to S3.", source: None } })))`.`
$ export AWS_REGION="garage"
$ ./quickwit run --config quickwit.yaml
✘ Command failed: Failed to connect to metastore: `Internal error: `Failed to get index files.` Cause: `StorageError(kind=InternalError, source=failed to construct request: no credentials in the property bag (ConstructionFailure(ConstructionFailure { source: SigningStageError { kind: MissingCredentials } })))`.`
$ export AWS_ACCESS_KEY_ID=foo
$ export AWS_SECRET_ACCESS_KEY=bar
$ ./quickwit run --config quickwit.yaml
...INFO quickwit_cluster::cluster: Joining cluster. ...
```

**Comment by guilload** [comment]
Yes, `access_key_id`, `secret_access_key`, and `region` are not honored at the moment. This will be fixed in Quickwit 0.6.2. In the meantime, one can use the "classic" AWS env vars for those. Namely, `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_REGION`.

**Comment by guilload** [comment]
@jpds, ping me directly on Discord if you need any help.

**Comment by guilload** [comment]
Fixed via #3590.
